lib/ocpp/v201/charge_point.cpp

```diff
@@ -148,6 +148,9 @@ ChargePoint::ChargePoint(const std::map<int32_t, int32_t>& evse_connector_struct
     v2g_certificate_expiration_check_timer([this]() { this->scheduled_check_v2g_certificate_expiration(); }),
     callbacks(callbacks),
     stop_auth_cache_cleanup_handler(false) {
+    EVLOG_info << "This ChargePoint constructor should no longer be used in favor of the one that allows for "
```

Comment:
Since we're relying on c++17, did you consider using the attribute `[[deprecated]]` (see https://en.cppreference.com/w/cpp/language/attributes/deprecated).  This warning you will get at compile time, instead of runtime and might be more appropriate w.r.t. API usage.

Reply:
Actually I wasn't sure if we had standardized on what to use to indicate deprecation. I have a question out in Zulip: https://lfenergy.zulipchat.com/#narrow/stream/417678-EVerest.3A-Framework-.26-Tools/topic/Deprecation.20procedure.3F

But I can make that change if that's what the project typically does. I am also not sure if the expectation is that the API/code is stable enough to require deprecation warnings or not.

Reply:
I am also not sure on what we would like to use here so would like to get this resolved before merging.

Otherwise the PR looks good.

Reply:
My preference would be to remove the old constructors entirely, but I only hesitate because I'm worried that someone is using them somewhere.

Reply:
I think that would be acceptable as long as its easy to migrate to the other constructors. There are breaking changes on a regular basis within libocpp so I think this one should be allowed as well.

@hikinggrass what do you think?

Reply:
@marcemmers @hikinggrass @a-w50 Since we don't currently have a deprecation strategy formally in place, and since this code base is not considered stable enough to worry about breaking changes, I'm inclined to remove the old constructors, which also allows for the ChargePoint class to be cleaned up a bit more. The main risk is if someone somewhere is using libocpp and the ChargePoint constructors. Other than changing documentation, I'm not sure how to communicate to anyone using the original constructors that they'll need to use the new one.

Any concerns with this approach?

Reply:
I think that's reasonable, like @marcemmers said there's still breaking changes from time to time at the moment anyways